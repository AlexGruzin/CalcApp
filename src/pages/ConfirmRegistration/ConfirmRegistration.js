import React from 'react';
import PropTypes from 'prop-types';
import { CONFIRMATION_STATUS } from 'constants/auth';
import Spinner from 'components/Spinner';
import { useTranslation } from 'react-i18next';

import './ConfirmRegistration.styles.less';

const ConfirmRegistration = ({ registrationStatus }) => {
  const { t } = useTranslation();

  return (
    <div className="confirm-reg">
      {!registrationStatus && (
        <>
          <Spinner />
          <div className="confirm-reg__text">{t('confirm:registration.processing')}</div>
        </>
      )}
      {registrationStatus === CONFIRMATION_STATUS.SUCCESS && (
        <div className="confirm-reg__done">{t('confirm:success')}</div>
      )}
      {registrationStatus === CONFIRMATION_STATUS.ERROR && (
        <div className="confirm-reg__block">{t('confirm:error')}</div>
      )}
    </div>
  );
};

ConfirmRegistration.propTypes = {
  registrationStatus: PropTypes.string,
};

ConfirmRegistration.defaultProps = {
  registrationStatus: '',
};

export default ConfirmRegistration;
