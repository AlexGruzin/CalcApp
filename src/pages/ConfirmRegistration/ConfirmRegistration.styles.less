.confirm-reg {
  display: flex;
  flex-grow: 1;
  background: var(--white);
  justify-content: center;
  align-items: center;
}

.confirm-reg__status {
  position: absolute;
}

.confirm-reg__text {
  margin-top: var(--space-x10);
  color: var(--font-black);
}

.confirm-reg .spinner > div {
  border-color: var(--black);
  border-right-color: transparent;
}

.confirm-reg__block {
  color: var(--error-color);
}

.confirm-reg__done {
  color: var(--turquoise);
}
