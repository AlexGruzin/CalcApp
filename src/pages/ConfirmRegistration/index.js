import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { getConfirmRegistrationStatus } from 'domains/auth/selectors';
import ConfirmRegistration from './ConfirmRegistration';

const selectors = createStructuredSelector({
  registrationStatus: getConfirmRegistrationStatus,
});

export default connect(selectors, null)(ConfirmRegistration);
