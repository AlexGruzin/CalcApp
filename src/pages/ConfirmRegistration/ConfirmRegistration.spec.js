import React from 'react';
import { render } from '@testing-library/react';
import ConfirmRegistration from './ConfirmRegistration';

describe('ConfirmRegistration tests', () => {
  it('Check snapshots of ConfirmRegistration are equal.', () => {
    const { asFragment } = render(<ConfirmRegistration />);
    expect(asFragment()).toMatchSnapshot();
  });
});
