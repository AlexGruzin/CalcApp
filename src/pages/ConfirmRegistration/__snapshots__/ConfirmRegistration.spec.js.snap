// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmRegistration tests Check snapshots of ConfirmRegistration are equal. 1`] = `
<DocumentFragment>
  <div
    class="confirm-reg"
  >
    <div
      class="spinner"
    >
      <div
        class="circle__Wrapper-sc-16bbsoy-0 dpSNSD"
        color="#fff"
        size="24"
      />
    </div>
    <div
      class="confirm-reg__text"
    >
      confirm:registration.processing
    </div>
  </div>
</DocumentFragment>
`;
